spec/javascripts/services/short_form_data_service_spec.coffee

```diff
@@ -8,37 +8,40 @@
     fakeSalesforceApplication = getJSONFixture('sample-salesforce-short-form.json')
     fakeApplication = getJSONFixture('sample-web-short-form.json')
     fakeApplicant = undefined
-    fakeListingDataService =
-      listing:
-        preferences: getJSONFixture('listings-api-listing-preferences.json').preferences
-      getPreference: jasmine.createSpy()
-      getPreferenceById: jasmine.createSpy()
-      RESERVED_TYPES:
-        VETERAN: 'Veteran'
-        DISABLED: 'Developmental disabilities'
-        SENIOR: 'Senior'
-      preferenceMap:
-        certOfPreference: "Certificate of Preference (COP)"
-        displaced: "Displaced Tenant Housing Preference (DTHP)"
-        liveWorkInSf: "Live or Work in San Francisco Preference"
-        liveInSf: "Live or Work in San Francisco Preference"
-        workInSf: "Live or Work in San Francisco Preference"
-        neighborhoodResidence: "Neighborhood Resident Housing Preference (NRHP)"
     fakeListingPreferenceService =
       hasPreference: ->
       getPreferenceById: ->
     fakeListingUnitService =
       listingHasReservedUnitType: ->
+    $state = undefined
+    $translate = {}
+    $localStorage = undefined
 
+    beforeEach module('ui.router')
+    # have to include http-etag to allow `$http.get(...).success(...).cached(...)` to work in the tests
+    beforeEach module('http-etag')
     beforeEach module('dahlia.services', ($provide) ->
-      $provide.value 'ListingDataService', fakeListingDataService
+      $provide.value '$translate', $translate
       $provide.value 'ListingPreferenceService', fakeListingPreferenceService
       $provide.value 'ListingUnitService', fakeListingUnitService
+      $provide.value 'ListingIdentityService', jasmine.createSpy()
+      $provide.value 'SharedService', jasmine.createSpy()
+      $provide.value 'ListingLotteryService', jasmine.createSpy()
+      $provide.value 'ExternalTranslateService', jasmine.createSpy()
+      $provide.value 'ListingEligibilityService', jasmine.createSpy()
+      $provide.value 'IncomeCalculatorService', jasmine.createSpy()
       return
     )
 
-    beforeEach inject((_ShortFormDataService_) ->
+    beforeEach inject((_ShortFormDataService_, _ListingDataService_, _ListingConstantsService_, _$state_, _$httpBackend_, _$localStorage_) ->
```

Comment:
Marçin finally helped me figure out how to pull real modules into tests. The syntax is terrible and you have to add all the dependencies above, but it's better than trying to constantly mock everything